MSRV check fails

From https://github.com/rp-rs/rp-hal/actions/runs/5933769438/job/16092767596?pr=670:
```
info: running `cargo build --examples --no-default-features --features critical-section-impl` on rp2040-hal (2/13)
error: package `indexmap v2.0.0` cannot be built because it requires rustc 1.64 or newer, while the currently active rustc version is 1.63.0
error: process didn't exit successfully: `/home/runner/.rustup/toolchains/1.63-x86_64-unknown-linux-gnu/bin/cargo build --examples --manifest-path rp2040-hal/Cargo.toml --no-default-features --features critical-section-impl` (exit status: 101)
Error: Process completed with exit code 1.
```

--- a/.github/workflows/build_and_test.yml
+++ b/.github/workflows/build_and_test.yml
@@ -82,7 +82,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: dtolnay/rust-toolchain@master
         with:
-          toolchain: 1.63
+          toolchain: 1.64
           target: thumbv6m-none-eabi
       - name: Install cargo-hack
         uses: baptiste0928/cargo-install@v2

--- a/rp2040-hal/CHANGELOG.md
+++ b/rp2040-hal/CHANGELOG.md
@@ -9,7 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### MSRV
 
-The Minimum-Supported Rust Version (MSRV) for this release is 1.63
+The Minimum-Supported Rust Version (MSRV) for this release is 1.64
 
 ### Fixed
 

--- a/rp2040-hal/Cargo.toml
+++ b/rp2040-hal/Cargo.toml
@@ -6,7 +6,7 @@ edition = "2021"
 homepage = "https://github.com/rp-rs/rp-hal"
 description = "A Rust Embeded-HAL impl for the rp2040 microcontroller"
 license = "MIT OR Apache-2.0"
-rust-version = "1.63"
+rust-version = "1.64"
 repository = "https://github.com/rp-rs/rp-hal"
 
 [package.metadata.docs.rs]

